src/pipe_block.rs

```diff
@@ -0,0 +1,131 @@
+//! TODO: Header
```

Comment:
Would be awesome to have a little blurb here about the big idea of how these pipe blocks are supposed to work.